core/src/cgcloud/core/box.py

```diff
@@ -319,18 +320,29 @@ def prepare( self, ec2_keypair_globs,
         if instance_type is None:
             instance_type = self.recommended_instance_type( )
 
-        virtualization_type = self.__get_virtualization_type( instance_type, virtualization_type )
+        virtualization_types = self.__get_virtualization_types( instance_type, virtualization_type )
 
+        image = None
         if image_ref is not None:
             image = self.__select_image( image_ref )
         else:
-            log.info( "Looking up default image for role %s, ... ", self.role( ) )
-            image = self._base_image( virtualization_type )
-            log.info( "... found %s.", image.id )
+            for type in virtualization_types:
+                log.info( "Looking up default image for role %s and type %s, ... ", self.role( ), type )
+                try:
+                    image = self._base_image( type )
+                except RuntimeError:
+                    log.info("... did not find a suitable image.")
+                else:
+                    log.info( "... found %s.", image.id )
+                if image is not None and image.virtualization_type in virtualization_types:
```

Comment:
The first term of this conjunction can be eliminated by moving the if statement into the else: branch of the try: block above.